myst_nbs/time_series/Forecasting_with_structural_timeseries.myst.md

```diff
@@ -0,0 +1,740 @@
+
+    fig, axs = plt.subplots(1, 3, sharex=False, figsize=(20, 6))
+    axs = axs.flatten()
+    for i, p in enumerate(percs[::-1]):
+        upper = np.percentile(
+            az.extract_dataset(idata_ar, group="prior_predictive", num_samples=100)["likelihood"],
+            p,
+            axis=1,
+        )
+        lower = np.percentile(
+            az.extract_dataset(idata_ar, group="prior_predictive", num_samples=100)["likelihood"],
+            100 - p,
+            axis=1,
+        )
+        color_val = colors[i]
+        axs[0].fill_between(
+            x=idata_ar["constant_data"]["t"],
+            y1=upper.flatten(),
+            y2=lower.flatten(),
+            color=cmap(color_val),
+            alpha=0.1,
+        )
+
+    axs[0].plot(
+        az.extract_dataset(idata_ar, group="prior_predictive", num_samples=100)["likelihood"].mean(
+            axis=1
+        ),
+        color="cyan",
+        label="Mean Prediction",
+    )
+
+    axs[0].scatter(x=idata_ar["constant_data"]["t"], y=idata_ar["constant_data"]["y"], color="k")
+    axs[0].set_title("Prior Predictive Fit", fontsize=20)
+
+    for i, p in enumerate(percs[::-1]):
+        upper = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"],
+            p,
+            axis=1,
+        )
+        lower = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"],
+            100 - p,
+            axis=1,
+        )
+        color_val = colors[i]
+        axs[2].fill_between(
+            x=idata_preds["predictions_constant_data"]["t"],
+            y1=upper.flatten(),
+            y2=lower.flatten(),
+            color=cmap(color_val),
+            alpha=0.1,
+        )
+
+        upper = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["yhat_fut"],
+            p,
+            axis=1,
+        )
+        lower = np.percentile(
+            az.extract_dataset(idata_preds, group="predictions", num_samples=100)["yhat_fut"],
+            100 - p,
+            axis=1,
+        )
+        color_val = colors[i]
+        axs[2].fill_between(
+            x=np.linspace(200, 250, 50),
+            y1=upper.flatten(),
+            y2=lower.flatten(),
+            color=cmap(color_val),
+            alpha=0.1,
+        )
+
+    axs[2].plot(
+        az.extract_dataset(idata_preds, group="predictions", num_samples=100)["likelihood"].mean(
+            axis=1
+        ),
+        color="cyan",
+    )
+    idata_preds.predictions.yhat_fut.mean(["chain", "draw"]).plot(ax=axs[2], color="cyan")
+    axs[2].scatter(x=idata_ar["constant_data"]["t"], y=idata_ar["constant_data"]["y"], color="k")
+    axs[2].set_title("Posterior Predictions Plotted", fontsize=20)
+    axs[2].axvline(np.max(idata_ar["constant_data"]["t"]), color="black")
+    az.plot_ppc(idata_ar, ax=axs[1])
+
+
+plot_fits(idata_ar, idata_preds)
```

Comment:
This one is tougher. I'll definitely add the legend for the cyan and black lines. I'll also use a mosaic pattern 

AABB
CCCC

to make the plot larger, like so: 
<img width="1130" alt="image" src="https://user-images.githubusercontent.com/10588167/198264265-186b23fb-3956-475c-ba94-9c79dca93936.png">

I'm very hesitant to add a colorbar or more labels to the plot for the colormap. The goal of the colormap is just to convey the graded probability of outcomes in the uncertainty interval and I think it does a good job of that as it is now. I think it would be overkill to add more labels mapping the colors to the range of probability.